system-parachains/coretime/coretime-kusama/src/migrations.rs

```diff
@@ -0,0 +1,225 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// The XCM Transact which was meant to set the leases as part of the Kusama relay runtime upgrade
+/// did not have enough weight. Therefore the leases were not migrated.
+///
+/// This migration populates the leases and restarts the sale from whichever timeslice it runs.
+///
+/// This does not affect storage structure, only values.
+pub mod bootstrapping {
+	use crate::{weights, Runtime, RuntimeOrigin};
+	use frame_support::{pallet_prelude::*, traits::OnRuntimeUpgrade};
+	#[cfg(feature = "try-runtime")]
+	use pallet_broker::{
+		AllowedRenewalId, AllowedRenewalRecord, AllowedRenewals, Configuration,
+		CoreAssignment::{Pool, Task},
+		CoreMask, LeaseRecordItem, SaleInfo, SaleInfoRecordOf, Schedule, ScheduleItem, Workplan,
+	};
+	use pallet_broker::{Leases, WeightInfo};
+	#[cfg(feature = "try-runtime")]
+	use sp_runtime::TryRuntimeError;
+	#[cfg(feature = "try-runtime")]
+	use sp_std::vec::Vec;
+
+	/// The log target.
+	const TARGET: &'static str = "runtime::bootstrapping::import-leases";
+
+	// Alias into the broker weights for this runtime.
+	type BrokerWeights = weights::pallet_broker::WeightInfo<Runtime>;
+
+	pub struct ImportLeases;
+
+	impl OnRuntimeUpgrade for ImportLeases {
+		fn on_runtime_upgrade() -> Weight {
+			// This migration contains hardcoded values only relevant to Kusama Coretime
+			// 1002000 before it has any leases. These checks could be tightened.
+			if Leases::<Runtime>::decode_len().unwrap_or(0) > 0 {
+				// Already has leases, bail
+				log::error!(target: TARGET, "This migration includes hardcoded values not relevant to this runtime. Bailing.");
+				return <Runtime as frame_system::Config>::DbWeight::get().reads(1);
+			}
+
+			for (para_id, end) in LEASES {
+				match pallet_broker::Pallet::<Runtime>::set_lease(
+					RuntimeOrigin::root(),
+					para_id,
+					end,
+				) {
+					Ok(_) =>
+						log::info!(target: TARGET, "Importing lease for parachain {}", &para_id),
+					Err(_) =>
+						log::error!(target: TARGET, "Importing lease for parachain {} failed!", &para_id),
+				}
+			}
+
+			// The values used in referendum 375 included 52 cores. Replaying this here shifts the
+			// start of the sale, while crucially populating the workplan with the leases and
+			// recalculating the number of cores to be offered. However, there are 4 system
+			// parachains + 1 pool core + 47 leases + 3 cores for the open market, therefore we need
+			// to start sales with 55 cores.
+			match pallet_broker::Pallet::<Runtime>::request_core_count(RuntimeOrigin::root(), 55) {
+				Ok(_) => log::info!(target: TARGET, "Request for 55 cores sent."),
+				Err(_) => log::error!(target: TARGET, "Request for 55 cores failed to send."),
+			}
+			match pallet_broker::Pallet::<Runtime>::start_sales(
+				RuntimeOrigin::root(),
+				5_000_000_000_0000,
```

Comment:
Wait, KSM has 12 decimals, it should be this:
```suggestion
				5_000_000_000_000,
```

Reply:
375:
<img width="632" alt="image" src="https://github.com/polkadot-fellows/runtimes/assets/25483142/df130bb5-178a-44cd-9346-e9f2e5cf02bf">

Reply:
Ah, sorry now I see what you were saying @eskimor